### ISSUE TITLE
Memory Leak

### ISSUE THREAD

**Issue by xealot** [gographics/imagick#40]
In `magick_wand_image.go` there appears to be a memory leak for `GetImageFormat()` and a few others. I am not that knowledgable about CGO but it seems like you need to free the pointer in this function and my testing proves that out.

The original function will push the memory usage every time its called.

```go
// Returns the format of a particular image in a sequence.
func (mw *MagickWand) GetImageFormat() string {
	return C.GoString(C.MagickGetImageFormat(mw.mw))
}
```

When I change the function to this (based on other functions in your code) the memory leak does not occur.

```go
// Returns the format of a particular image in a sequence.
func (mw *MagickWand) GetImageFormat() string {
	p := C.MagickGetImageFormat(mw.mw)
	defer C.free(unsafe.Pointer(p))
	return C.GoString(p)
}
```

There appears to be multiple examples of this in `magick_wand_image.go`, `pixel_wand.go` and `magick_wand.go` but I don't know enough to know which ones should be that way and which should not.

**Comment by xealot** [comment]
I also found this which helped lead me to this:
http://www.imagemagick.org/discourse-server/viewtopic.php?t=21724

**Comment by phacops** [comment]
Nice catch. You should do a pull request for this.

**Comment by justinfx** [comment]
This looks related to pull request #39, which I plan to look at merging today. It is using relinquish instead of free, in areas that needed to release memory.

**Comment by xealot** [comment]
Awesome, well I also made a pull request that fixes things that seemed to match that pattern. Don't know if I did the right thing in all cases but please take a look.

**Comment by justinfx** [comment]
Thanks! Ok I just merged #39 , which added a private `relinquishMemory()` and `relinquishMemoryCStringArray()`. And you have found more areas where the freeing has not been properly done.

http://www.imagemagick.org/discourse-server/viewtopic.php?t=7523

Can you update your pull request branch to use `relinquishMemory()` based on master?

**Comment by xealot** [comment]
See if that has what you need.
